yew-macro/src/html_tree/html_if.rs

```diff
@@ -0,0 +1,128 @@
+use super::HtmlRoot;
+use crate::PeekValue;
+use boolinator::Boolinator;
+use proc_macro2::{Delimiter, TokenStream};
+use quote::{quote, quote_spanned, ToTokens};
+use syn::buffer::Cursor;
+use syn::parse::{Parse, ParseStream, Result as ParseResult};
+use syn::spanned::Spanned;
+use syn::{braced, token, Expr, Token};
+
+pub struct HtmlIf {
+    if_token: Token![if],
+    cond: Box<Expr>,
+    then_branch: HtmlBranch,
+    else_branch: Option<(Token![else], Box<HtmlBranchOrIf>)>,
+}
+
+impl PeekValue<()> for HtmlIf {
+    fn peek(cursor: Cursor) -> Option<()> {
+        let (ident, _) = cursor.ident()?;
+        (ident == "if").as_option()
+    }
+}
+
+impl Parse for HtmlIf {
+    fn parse(input: ParseStream) -> ParseResult<Self> {
+        let if_token = input.parse()?;
+        let cond = input.parse()?;
+        let then_branch = input.parse()?;
+        let else_branch = input
+            .parse::<Token![else]>()
+            .ok()
+            .map(|else_token| input.parse().map(|branch| (else_token, branch)))
+            .transpose()?;
+
+        Ok(HtmlIf {
+            if_token,
+            cond,
+            then_branch,
+            else_branch,
+        })
+    }
+}
+
+impl ToTokens for HtmlIf {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let HtmlIf {
+            if_token,
+            cond,
+            then_branch,
+            else_branch,
+        } = self;
+        let default_else_branch = syn::parse_str("{}").unwrap();
+        let else_branch = else_branch
+            .as_ref()
+            .map(|(_, branch)| branch)
+            .unwrap_or(&default_else_branch);
+        let new_tokens = quote_spanned! {if_token.span()=>
+            if #cond #then_branch else #else_branch
+        };
+
+        tokens.extend(new_tokens);
+    }
+}
+
+pub struct HtmlBranch {
```

Comment:
I think I already mentioned my distaste for the meaningless "Html" prefix. I'll leave it up to you but don't feel like you need to stick to the "Html" prefix just because it's used in the other files (I suspect the prefix is only there for historical reasons).
At some unknown point in the future I would like to finally get rid of it everywhere.

Reply:
haha you probably did :sweat_smile: I'm so sorry. I will fix it in a future PR if you don't mind (I have 2 mores (for the `for` and for the `match` so....) and also I really want this if for 0.18 if possible because it would un-bloat yewprint a lot)

Reply:
oof, do you ever read an old comment and immediately cringe? I'm sure it wasn't my intention but my comment really comes across way too angry :D

Anyway, I'm fine with fixing it in another PR; the more the merrier.

Reply:
I want to tell you I understand and everything is okay :pray:  .... but I don't remember much about this conversation. I feel like I'm a third person now.

Ahh I see. This will fit right into the issue about cleaning things. Great!